### ISSUE TITLE
Consider Changing Public API

### ISSUE THREAD

**Issue by pburtchaell** [pburtchaell/redux-promise-middleware#219]
## Problem

When applying the middleware to my Redux store, I need to first call the `promiseMiddleware()` method, which returns the middleware. 

```js
import promise from 'redux-promise-middleware';

const store = createStore(reducers, null, applyMiddleware(
  thunk,
  promiseMiddleware(),
  logger,
));
```

We originally implemented the middleware in this way to support custom configuration.

```js
import promise from 'redux-promise-middleware';

const store = createStore(reducers, null, applyMiddleware(
  thunk,
  promiseMiddleware({
    delimiter: '/'
  }),
  logger,
));
```

However, after using the middleware for several years now, I'm finding this implementation is unncessarily verbose. It's also diifferent from other Redux middleware APIs.

Yes, the API should support custom configuration, but I also think we could simplify the API.

## Solution(s)

Default API: 

```js
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import logger from 'redux-logger';

const store = createStore(reducers, null, applyMiddleware(
  thunk,
  promise,
  logger,
));
```

Custom middleware API: 

```js
import thunk from 'redux-thunk';
import { createPromise } from 'redux-promise-middleware';
import { createLogger } from 'redux-logger';

const store = createStore(reducers, null, applyMiddleware(
  thunk,
  createPromise(myConfigObject),
  createLogger(myLoggerConfigObject),
));
```

**Comment by zhanyuzhang** [comment]
Maybe you should check the arguments first. Just as follows:
```
export default function promiseMiddleware(config = {}) {
  let { dispatch, promiseTypeSuffixes, promiseTypeDelimiter } = config;
  const PROMISE_TYPE_SUFFIXES = promiseTypeSuffixes || defaultTypes;
  const PROMISE_TYPE_DELIMITER = promiseTypeDelimiter || '_';

  const fn = next => action => {
    // the main code comming here....
  }

  // if 'dispatch' in the 'config' object, return the main code directly.
  if(dispatch) {
    return fn;
  }

  // else return the middleware function
  return ref => {
    dispatch = ref.dispatch;
    return fn;
  }
}
```

**Comment by pburtchaell** [comment]
## Update

There is an open PR by @zhanyuzhang at #223.

**Comment by pburtchaell** [comment]
Yes! Feel free to make a PR. Maybe use #223 as a starting point—had some feedback on that PR and it was never addressed, so I closed it.

**Comment by pburtchaell** [comment]
## Update

Thanks to a PR by @rahulbdominic and initial from @zhanyuzhang, we have this on `master`. Will work on a major version release soon.

**Comment by pburtchaell** [comment]
Closed with release of version 6.0.0.
